statR version 2.4.0 (#60)
statistikZH/statR: vignettes/Excel-Tabellen.Rmd

  collapse = TRUE,
  comment = "#>"
)

library(yaml)
library(statR)
library(dplyr)
library(ggplot2)
```

## Allgemeine Informationen

> Das `statR` Package enthält Funktionen zur schnellen Erzeugung von Microsoft Excel-Dateien mithilfe des [openxlsx](https://github.com/awalker89/openxlsx) Pakets. Die Voreinstellungen orientieren sich am Corporate Design vom Statistischen Amt Kanton Zürich, Anwender können diese Voreinstellungen aberauch an ihre Bedürfnisse anpassen. Entweder, indem sie Anpassungen in den Eingabeargumenten der Funktionen vornehmen, oder indem sie eine eigene Konfiguration anlegen.

Reviewer: ```suggestion
Das `statR` Package enthält Funktionen zur schnellen Erzeugung von Microsoft Excel-Dateien mithilfe des [openxlsx](https://github.com/awalker89/openxlsx) Pakets. Die Voreinstellungen orientieren sich am Corporate Design vom Statistischen Amt Kanton Zürich, Anwender können diese Voreinstellungen aber auch an ihre Bedürfnisse anpassen. Entweder, indem sie Anpassungen in den Eingabeargumenten der Funktionen vornehmen, oder indem sie eine eigene Konfiguration anlegen.
```
Fix Typo

```{r, echo = TRUE, eval = FALSE}
mtcars_title <- "Motor trend car road tests"
mtcars_source <- paste0("Henderson and Velleman (1981). Building multiple ",
                        "regression models interactively. ",
                        "Biometrics, 37, 391–411.")
mtcars_metadata <- paste0("The data was extracted from the 1974 Motor Trend ",
                          "US magazine and comprises fuel consumption and ",
                          "10 aspects of automobile design and performance ",
                          "for 32 automobiles (1973–74 models).")
aXLSX(data = mtcars,
      file = "output/axlsx_demo",
      title = mtcars_title,
      source = mtcars_source,
      metadata = mtcars_metadata,
      author = "Vorname Nachname",
      config = "default")
```

![Screenshot des Datenblatts in der Datei motor_trend_car_road_tests.xlsx](./img/aXLSX_1.PNG){width=95%}

![Screenshot des Metadatenblatts der Datei motor_trend_car_road_tests.xlsx](./img/aXLSX_2.PNG){width=95%}

### splitXLSX

> Mit der Funktion \code{splitXLSX} lässt sich ein beliebiger Datensatz anhand eines Merkmals auf mehrere Arbeitsblätter verteilen. Zur Vereinfachung der Navigation wird im ersten Arbeitsblatt ein Inhaltsverzeichnis mit Hyperlinks angelegt. Die Funktion eignet sich besonders für Variabl besonders geeignet in Fällen, wenn die Variable eine begrenzte Anzahl von Ausprägungen mit intuitiven Bedeutungen hat (zB. Altersklassen, Jahre oder Regionen). 

Reviewer: ```suggestion
Mit der Funktion \code{splitXLSX} lässt sich ein beliebiger Datensatz anhand eines Merkmals auf mehrere Arbeitsblätter verteilen. Zur Vereinfachung der Navigation wird im ersten Arbeitsblatt ein Inhaltsverzeichnis mit Hyperlinks angelegt. Die Funktion ist besonders geeignet in Fällen, wenn die Variable eine begrenzte Anzahl von Ausprägungen mit intuitiven Bedeutungen hat (zB. Altersklassen, Jahre oder Regionen). 
```
Fix Wording

Implizit wird angenommen, dass die Quellangaben und Metadaten für jede der Ausprägungen gelten. Andernfalls sollte die Funktion \code{datasetsXLSX} verwendet werden.

Im folgenden Beispiel wird aus dem \code{mtcars} Datensatz eine Arbeitsmappe erstellt, in der die Datenpunkte entsprechend der Anzahl von Zylindern auf mehrere Arbeitsblätter aufgeteilt werden.
```{r,echo = TRUE, eval = FALSE}
splitXLSX(data = mtcars,
          file = "output/splitxlsx_demo",
          sheetvar = "cyl",
          title = mtcars_title,
          source = mtcars_source,
          metadata = mtcars_metadata,
          author = "Vorname Nachname",
          config = "default"
)
```

### datasetsXLSX

> Die Funktion \code{datasetsXLSX} ermöglicht die Erzeugung komplexerer Arbeitsmappen mit mehreren Datensätzen sowie die Einbindung von Grafiken (\code{ggplot}-Objekte oder Pfadangaben zu Bild-Dateien). Die Datensätze und Grafiken werden dabei als Liste im Funktionsargument \code{datasets} übergeben. Jeder Eintrag wird in ein neues Arbeitsplatt exportiert und kann wahlweise mit Titeln, Quellangaben und Metadaten versehen werden. Wie in \code{splitXLSX} wird ein Deckblatt mit einem Inhaltsverzeichnis erzeugt. Ferner besteht die Möglichkeit, mit dem Argument \code{metadata_sheet} ein Beiblatt zu erzeugen.

Reviewer: ```suggestion
Die Funktion \code{datasetsXLSX} ermöglicht die Erzeugung komplexerer Arbeitsmappen mit mehreren Datensätzen sowie die Einbindung von Grafiken (\code{ggplot}-Objekte oder Pfadangaben zu Bild-Dateien). Die Datensätze und Grafiken werden dabei als Liste im Funktionsargument \code{datasets} übergeben. Jeder Eintrag wird in ein neues Arbeitsblatt exportiert und kann wahlweise mit Titeln, Quellangaben und Metadaten versehen werden. Wie in \code{splitXLSX} wird ein Deckblatt mit einem Inhaltsverzeichnis erzeugt. Ferner besteht die Möglichkeit, mit dem Argument \code{metadata_sheet} ein Beiblatt zu erzeugen.
```
Fix Typo

  sheetname = list("mtcars", "PlantGrowth", "Grafik1", "Grafik2"),
  title = list(mtcars_title, PlantGrowth_title, grafik1_title, grafik2_title),
  source = list(mtcars_source, PlantGrowth_source, grafik1_source, grafik2_source),
  metadata = list(mtcars_metadata, PlantGrowth_metadata, NULL, NULL),
  grouplines = NULL,
  group_names = NULL,
  plot_width = 10,
  plot_height = list(10, 5),
  auftrag_id = NULL,
  author = "Vorname Nachname",
  metadata_sheet = beiblatt,
  overwrite = TRUE,
  config = "default")
```

Das Beispiel weist aus Illustrationsgründen zwei Besonderheiten auf. Erstens werden im Beispielscode im Argument \code{metadata} zwei Einträge als NULL gesetzt. Dies entspricht einer Leerlassung. Zweitens wird für das Argument \code{plot_width} nur ein Wert definiert. Dieser Wert wird für alle weiteren Grafiken verwendet. Im Argument \code{plot_height} werden die Höhen für beide Plots definiert. Wie die Länge des Beispielscodes bereits impliziert, kann ein \code{datasetsXLSX}-Aufruf schnell unübersichtlich werden. Ferner werden bei Änderungen in der Reihenfolge  entsprechende Anpassungen bei bis zu 9 Listen fällig. 

> Mit den Funktionen der \code{add_attribute}-Familie wird eine alternative Möglichkeit angeboten, \code{datasetsXLSX} zu parametrisieren. Jedes Eingabeobjekt (ob Daten oder Grafiken) kann mit Attributen für Titel, Quellen, Metadaten, und mehr versehen werden. Das so augmentierte Objekt kann ohne weitere Angaben an \code{datasetsXLSX} übergeben werden. Schritt kann einfach Philosophie unterstützt, dass Daten und Datenbeschreibung \code{add_sheetname}, \code{add_title}, \code{add_source}, etc. an. Am letzten Beispiel illustriert:

Reviewer: Absatz auf Grammatik prüfen: "Philosophie"

  add_metadata(mtcars_metadata)

PlantGrowth_dataset <- PlantGrowth |>
  add_sheetname("PlantGrowth") |> 
  add_title(PlantGrowth_title) |>
  add_source(PlantGrowth_source) |>
  add_metadata(PlantGrowth_metadata)

# --- grafik1 ---
grafik1 <- grafik1 |>
  add_sheetname("Grafik1") |>
  add_title("mtcars: weight vs engine power (hp)") |>
  add_source(mtcars_source) |>
  add_plot_size(c(10, 10))

# --- grafik2 ---
grafik2 <- grafik2 |>
  add_sheetname("Grafik2") |>
  add_title("mtcars: weight vs engine displacement (cu. in.)") |>
  add_source(mtcars_source) 

# --- Export .xlsx ---
datasetsXLSX(
  file = "output/datasetsXLSX_demo2.xlsx",
  datasets = list(mtcars_dataset, PlantGrowth_dataset, grafik1, grafik2),
  author = "Vorname Nachname",
  metadata_sheet = beiblatt,
  overwrite = TRUE,
  config = "default")

```

![Screenshot: Titelblatt der Datei datasetsXLSX_demo.xlsx](./img/datasetsXLSX_1.PNG){width=95%}

![Screenshot: Tabelle in 2. Arbeitsblatt der Datei datasetsXLSX_demo.xlsx](./img/datasetsXLSX_2.PNG){width=95%}

![Screenshot: Grafik in 5. Arbeitsblatt der Datei datasetsXLSX_demo.xlsx](./img/datasetsXLSX_3.PNG){width=95%}

## Ausgewählte Themen

### Benutzerkonfigurationen

> Um das Paket einem breiteren Anwenderkreis zugänglich zu machen, wurden diverse Aspekte im Design parametrisiert. Textfelder wie Kontaktinformationen können in den Funktionen direkt übersteuert werden. Für längerfristige Anpassungen besteht die Möglichkeit, Nutzerkonfigurationen anzulegen. Die Konfigurationen sind YAML-Dateien, in denen Einträge durch Schlüsselwörter mit zugehörigen Werten hinterlegt sind. 

Reviewer: Verständlicher formulieren. YAML beschreiben oder Link. Was bedeutet übersteuern konkret?

Author: Im Output hat es einen Print von default.yaml. Den Abschnitt habe ich aber noch so überarbeite, dass klar ist, was mit übersteuern gemeint ist.
